### PR TITLE
Add readonly repository viewset to the plugin API

### PR DESCRIPTION
**Pull request by goosemania**

### CODE REVIEW

**Comment by mdellweg** [comment]
Once you have the commit machinery pass the linting stage and all the CI green, i'd be happy to approve.

File: pulpcore/app/viewsets/repository.py [hunk 1]
```diff
@@ -43,15 +43,13 @@ class Meta:
         fields = {"name": NAME_FILTER_OPTIONS}
 
 
-class ImmutableRepositoryViewSet(
+class ReadOnlyRepositoryViewSet(
     NamedModelViewSet,
     mixins.CreateModelMixin,
```
Comment (ipanova) [inline pulpcore/app/viewsets/repository.py:48]: I bet this should not be here ;)

File: pulpcore/app/viewsets/repository.py [hunk 2]
```diff
@@ -43,15 +43,13 @@ class Meta:
...
     mixins.RetrieveModelMixin,
-    mixins.ListModelMixin,
```
Comment (ipanova) [inline pulpcore/app/viewsets/repository.py:50]: We should probably leave the 'list'

File: pulpcore/app/viewsets/repository.py (hunk 1 above)
Comment (goosemania) [reply pulpcore/app/viewsets/repository.py]: It was the typo, I fixed it. maybe at the same time you were reviewing.
As a reminder, it's a draft PR, I suggest not to spend time on it until it moves to a normal one.

**Review by daviddavis — APPROVED** [review]

**Review by ipanova — APPROVED** [review]

**Pull request merged**